### PR TITLE
Fix tx pool inconsistent when receive duplicated hash txs.

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

File: tx-pool/src/component/container.rs [hunk 1]
```diff
@@ -154,6 +154,12 @@ impl SortedTxMap {
     pub fn add_entry(&mut self, mut entry: TxEntry) -> Option<TxEntry> {
         let short_id = entry.transaction.proposal_short_id();
 
+        let removed_entry = if self.contains_key(&short_id) {
```
Comment (u2) [inline tx-pool/src/component/container.rs]: I think if there is tx_hash collision, it should return None.

Comment (jjyr) [reply tx-pool/src/component/container.rs]: Solve tx_hash collision is not a responsibility for this struct, we need to handle collision in the pool logic, here the struct only guarantees internal consistency.

The behavior is just like https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.insert

Comment (u2) [reply tx-pool/src/component/container.rs]: ok

**Comment by quake** [comment]
bors r=quake,u2,zhangsoledad

**Pull request merged**